### PR TITLE
Ee 283

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/controllers/shipyard/ApprenticeStorekeeperController.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        (formWithErrors: Form[_]) =>
+          Future.successful(BadRequest(view(formWithErrors, mode))),
+
+        value => {
+          for {
+            updatedAnswers <- if (value) {
+              Future.fromTry(request.userAnswers.set(AluminiumApprenticePage, value)
```
Comment (MatthewAlexanderBeck) [inline app/controllers/shipyard/ApprenticeStorekeeperController.scala]: userAnswers.set(AluminiumApprenticePage) should this not be ApprenticeStoreKeeperPage?

File: app/controllers/shipyard/ApprenticeStorekeeperController.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                .flatMap(_.set(ClaimAmount, ClaimAmounts.Shipyard.apprentice))
+              )
+            } else {
+              Future.fromTry(request.userAnswers.set(AluminiumApprenticePage, value)
```
Comment (MatthewAlexanderBeck) [inline app/controllers/shipyard/ApprenticeStorekeeperController.scala]: Same here

File: app/controllers/shipyard/ApprenticeStorekeeperController.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                .flatMap(_.set(ClaimAmount, ClaimAmounts.Shipyard.allOther))
+              )
+            }
+            _ <- sessionRepository.set(request.identifier, updatedAnswers)
+          } yield Redirect(navigator.nextPage(AluminiumApprenticePage, mode)(updatedAnswers))
```
Comment (MatthewAlexanderBeck) [inline app/controllers/shipyard/ApprenticeStorekeeperController.scala]: And here

File: test/controllers/shipyard/ApprenticeStorekeeperControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        .build()
+
+      val request = FakeRequest(POST, apprenticeStorekeeperRoute)
+        .withFormUrlEncodedBody(("value", "false"))
+
+      val result = route(application, request).value
+
+      val userAnswers2 = userAnswers
+        .set(ClaimAmount, ClaimAmounts.Shipyard.allOther).success.value
+        .set(AluminiumApprenticePage, false).success.value
```
Comment (MatthewAlexanderBeck) [inline test/controllers/shipyard/ApprenticeStorekeeperControllerSpec.scala]: .set(AluminiumApprenticePage, false).success.value?

File: test/controllers/shipyard/ApprenticeStorekeeperControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        .build()
+
+      val request = FakeRequest(POST, apprenticeStorekeeperRoute)
+        .withFormUrlEncodedBody(("value", "true"))
+
+      val result = route(application, request).value
+
+      val userAnswers2 = userAnswers
+        .set(ClaimAmount, ClaimAmounts.Shipyard.apprentice).success.value
+        .set(AluminiumApprenticePage, true).success.value
```
Comment (MatthewAlexanderBeck) [inline test/controllers/shipyard/ApprenticeStorekeeperControllerSpec.scala]: .set(AluminiumApprenticePage, true).success.value?

File: app/pages/shipyard/ApprenticeStorekeeperPage.scala [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package pages.shipyard
+
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object ApprenticeStorekeeperPage extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (MatthewAlexanderBeck) [inline app/pages/shipyard/ApprenticeStorekeeperPage.scala]: We store Industries in a sub directory called "Industry", have a look at other industry pages for how this is done

File: app/controllers/shipyard/ApprenticeStorekeeperController.scala [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.shipyard.ApprenticeStoreKeeperView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ApprenticeStorekeeperController @Inject()(
+                                         override val messagesApi: MessagesApi,
+                                         sessionRepository: SessionRepository,
+                                         @Named("Generic") navigator: Navigator,
```
Comment (vinnicombe) [inline app/controllers/shipyard/ApprenticeStorekeeperController.scala]: Is this the correct navigator? And use NavConstants.

File: test/generators/PageGenerators.scala [hunk 8]
```diff
@@ -31,10 +31,16 @@ import pages.foodCatering._
 import pages.heating._
 import pages.security._
 import pages.printing._
-import pages.shipyard.ShipyardOccupationList1Page
+import pages.shipyard.{ApprenticeStorekeeperPage, ShipyardOccupationList1Page, ShipyardOccupationList2Page}
```
Comment (vinnicombe) [inline test/generators/PageGenerators.scala]: Can you _ these imports for consistency please

File: test/generators/UserAnswersGenerator.scala [hunk 9]
```diff
@@ -33,7 +33,7 @@ import pages.manufacturing._
 import pages.police._
 import pages.printing._
 import pages.security._
-import pages.shipyard.ShipyardOccupationList1Page
+import pages.shipyard.{ApprenticeStorekeeperPage, ShipyardOccupationList1Page, ShipyardOccupationList2Page}
```
Comment (vinnicombe) [inline test/generators/UserAnswersGenerator.scala]: Can you _ these imports for consistency please

**Pull request merged**